### PR TITLE
DEVEX-721 wait for release to be readable to avoid commit not found e…

### PR DESCRIPTION
**Pull request by grosser**
…rrors

@zendesk/samson @bquorning @benhass @dadah89 @anthonywoo @clairekrogers

### CODE REVIEW

File: app/models/release_service.rb [hunk 1]
```diff
@@ -21,8 +22,14 @@ def can_release?
...
   private
 
-  def push_tag_to_git_repository(release)
-    GITHUB.create_release(@project.repository_path, release.version, target_commitish: release.commit)
+  def push_tag_to_git_repository(version, commit)
+    GITHUB.create_release(@project.repository_path, version, target_commitish: commit)
+  end
+
+  def ensure_tag_in_git_repository(tag)
+    Samson::Retry.with_retries [Octokit::NotFound], 3, sleep: 1 do
```
Comment (craig-day) [inline app/models/release_service.rb]: is 3 enough? or is 1 second enough? Maybe make ENV configurable for easier adjustment?

File: lib/samson/retry.rb [hunk 2]
```diff
@@ -5,12 +5,16 @@ def self.retry_when_not_unique(&block)
...
 
-    def self.with_retries(errors, count, options = nil)
-      check = options && options[:if]
+    def self.with_retries(errors, count, only_if: nil, sleep: nil)
       yield
     rescue *errors
       count -= 1
-      count >= 0 && (!check || check.call($!)) ? retry : raise
+      if count >= 0 && (!only_if || only_if.call($!))
+        sleep(sleep) if sleep
```
Comment (craig-day) [inline lib/samson/retry.rb]: super confusing variable names here, maybe name the argument `sleep_duration` or `wait_time` or something?

File: app/models/release_service.rb (hunk 1 above)
Comment (grosser) [reply app/models/release_service.rb]: I'd bet that's enough ... we currently do the new deploy immediately so the slightest wait would fix it

File: lib/samson/retry.rb (hunk 2 above)
Comment (grosser) [reply lib/samson/retry.rb]: looks so neat though ... changed 😿

**Pull request merged**